### PR TITLE
Fixing Ninja warning about multiple generate liblz4.a and adding missing cherry-pick

### PR DESCRIPTION
**Pull request by oshadura**

### CODE REVIEW

**Comment by oshadura** [comment]
@amadio can you please take a look?

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -115,8 +115,5 @@ endif()
 if(builtin_lzma)
   ROOT_ADD_BUILTIN_DEPENDENCIES(Core LZMA)
 endif()
-if(builtin_lz4)
-  ROOT_ADD_BUILTIN_DEPENDENCIES(Core LZ4)
-endif()
```
Comment (amadio) [inline core/CMakeLists.txt]: Why is this removed? What happens when LZ4 becomes the default? Won't this break `Core`?

Comment (oshadura) [reply core/CMakeLists.txt]: I added these lines before as a part of integration of LZ4, my fault that i didn't check what is actually doing ROOT_ADD_BUILTIN_DEPENDENCIES. Now I rechecked and it is generating adding custiom target LZ4LIBS (liblz4) while I actually building lz4 before, and if to check libCore.so linking, it is also trying to link LZ4LIBS together with liblz4.a as a LZ4 target.

It is output of duplicated rules from Ninja for lz4: 
```build core/LZ4LIBS: phony core/CMakeFiles/LZ4LIBS lib/liblz4.a LZ4# Phony custom command for lib/liblz4.a build lib/liblz4.a: phony || LZ4build core/CMakeFiles/LZ4LIBS: phony lib/liblz4.a || LZ4 ```

Comment (amadio) [reply core/CMakeLists.txt]: Ok, thanks for the explanation!

**Comment by amadio** [comment]
Just for those curious, this is the 6.08 branch, so it does not yet have the IMT features from the failing tests, hence the failures.
